spec/schema.json

```diff
@@ -410,6 +416,23 @@
         "SUBMITTED",
         "PROCESSED"
       ]
+    },
+    "Ineligible": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "$ref": "#/definitions/IneligibilityStatus"
+        }
+      }
+    },
+    "IneligibilityStatus": {
+      "type": "string",
+      "enum": [
+        "INELIGIBLE_COMPANY_STRUCK_OFF",
```

Comment:
I thought we agreed in the meeting that what we'd return was a code to the web to allow for the screen navigation (not the finer-grained action codes in the story) - so we'd only need two codes in theory, INELIGIBLE_COMPANY_STRUCK_OFF for OBJ-208 and (probably) INELIGIBLE_NO_DISSOLUTION_ACTION for OBJ-209 if we're making the spec change for that story too now.

Reply:
It's not something we specified on the ticket so I think these were just ideas we were playing with. The logic would be much the same in the web because we would look for "INELIGIBLE_COMPANY_NO_ELIGIBILITY_CODE" in which case we redirect to the OBJ-209 screen and if not then we go to the OBJ-208 screen. Having these codes would allow us to log or provide a reason on the 208 error page should that become a requirement.

Reply:
Worth checking with @markpit as he wrote up the sub-task. I thought we only bothered adding the ineligible code to the 400 response in order to distinguish between the 208 and 209 stories, in which case two codes would be sufficient. I guess it's where we want that business logic around the action codes to live - in the web or the API... (the finer-grained action code should definitely be logged in the API though)

Reply:
I thought it would be whether the action code was in one of the groups specified in the story. The api would get the action code, check what group it was in and return the status enum value for that ineligible group. That status would also be applied to the database record and the action code stored on it as well.
   If we don't group them, if more codes are added/subtracted in future we would need to update the spec and create new enum value for each ineligible action code and modify the web app to look for that new status. I think if we go with the groups approach it will give us more flexibility if they do want to add/subtract codes to a group and at this point the web is on the only user of the api (which needs 2 codes to show 2 ineligible screens)

Reply:
Any thoughts on what these groups should be? Here's a suggestion:

ELIGIBLE_COMPANY
INELIGIBLE_COMPANY
NO_ACTION_FOUND

Reply:
i'd go with what Martin said and have 2 statuses to indicate the 2 groups of ineligible codes, perhaps go with martin's suggestions above for the statuses (or similar)

Reply:
if it's eligible, we would return a 200 ok http code and continue as normal (so no need to return a status if 200 ok)

Reply:
... and these are the codes we should add to the Objection model for OBJ-230?

Reply:
Gone with Martin's suggestion and grouped them.

Reply:
We'd need them added to the ObjectionStatusEnum which you can do as part of OBJ-230 and we need a new field on Objection called actionCode to store the actual action code against the objection. We's also need to update the builders for actionCode as well.

Reply:
So we store both the CHIPS code and our group codes on the object?

Reply:
yes we'll store both although the objection status field already exists, we'll just populate it with the new values. And add a new action code field.